include/Surelog/Common/PathId.h

```diff
@@ -22,6 +38,19 @@
 #endif
 
 namespace SURELOG {
+/**
+ * class PathId
+ *
+ * Used to uniquely represent a file or directory abstractly.
```

Comment:
Does it represent a file/directory or a name of such ? It sounds like more a name which then can be used to open something that conceptually is a stream or directory.